### PR TITLE
Added migration file to OpenVINO 2023.3.0

### PR DESCRIPTION
**Pull request by ilya-lavrenov**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/migrations/libopenvino_dev202330.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+__migrator:
+  build_number: 1
+  commit_message: Rebuild for libopenvino-dev 2023.3.0
+  kind: version
+  migration_number: 1
+migrator_ts: 1705841975.2201185
+libopenvino-dev:
```
Comment (ilya-lavrenov) [inline recipe/migrations/libopenvino_dev202330.yaml:7]: - not sure what is proper name here `libopenvino_dev` or `libopenvino-dev` (actual package name)
- conda bot has not created automatically migration file for OpenVINO, do you know why?

Comment (xhochy) [reply recipe/migrations/libopenvino_dev202330.yaml:7]: This needs to be an underscore. `-` are not allowed for the key here.

Comment (ilya-lavrenov) [reply recipe/migrations/libopenvino_dev202330.yaml]: thanks @xhochy, fixed
do you know answer on the second question?

Comment (xhochy) [reply recipe/migrations/libopenvino_dev202330.yaml]: No.

**Pull request merged**